client/src/components/Ticketing/ticketingmanager/Season/components/SeasonEvents.tsx

```diff
@@ -1,7 +1,229 @@
+    delete eventToAdd['numshows'];
+    delete eventToAdd['description'];
+    delete eventToAdd['title'];
+
+    const updateEventCall = await updateEventSeason(eventToAdd, token);
+    setShowPopUp(true);
+    if (updateEventCall) {
+      setAllEventInfo([...allEventInfo, eventToAddCopy]);
+      setEventsNotInSeason(updatedEventsNotInSeason);
+      setPopUpMessage({
+        title: 'Success',
+        message: 'The event has been added to the season!',
+        success: true,
+        handleClose: () => setShowPopUp(false),
+        handleProceed: () => setShowPopUp(false),
+      });
+    } else {
+      setPopUpMessage({
+        title: 'Error',
+        message:
+          'There was a error adding the event to the season. Pleas try again.',
+        success: false,
+        handleClose: () => setShowPopUp(false),
+        handleProceed: () => setShowPopUp(false),
+      });
+    }
+  };
+
+  useEffect(() => {
+    void handleGetAllEvents();
+  }, []);
+
+  return (
+    <div className='rounded-xl p-7 bg-white text-lg mt-5'>
+      <section className='flex flex-col gap-4 items-center mb-6 tab:flex-row tab:justify-center tab:flex-wrap min-[1076px]:justify-between'>
+        <article className='flex flex-wrap gap-2'>
+          <h1 className='text-3xl'>Season Events </h1>
+          <button
+            className='flex gap-1 items-center bg-green-500 hover:bg-green-700 disabled:bg-gray-500 text-white p-2 rounded-xl'
+            disabled={isFormEditing || isAddEventActive}
+            onClick={() => setIsAddEventActive(true)}
+          >
+            <svg
+              xmlns='http://www.w3.org/2000/svg'
+              fill='none'
+              viewBox='0 0 20 20'
+              strokeWidth='3'
+              stroke='currentColor'
+              className='w-5 h-5 mb-1'
+            >
+              <path
+                strokeLinecap='round'
+                strokeLinejoin='round'
+                d='M12 6v12m6-6H6'
+              />
+            </svg>
+            <p>Add event</p>
+          </button>
+        </article>
+        <article>
+          <label className='pr-2' htmlFor='eventSelect'>
+            Sort by:
+          </label>
+          <select
+            name='sortBy'
+            id='eventSelect'
+            className='p-1 border border-zinc-500 rounded-lg'
+          >
+            <option value='eventName'>Event Name - Ascending</option>
+          </select>
+        </article>
+      </section>
+      {isAddEventActive && (
+        <div className='h-96 overflow-scroll mb-3'>
+          <h2 className='text-2xl mb-2 inline'>Unassigned Events</h2>
+          <button
+            onClick={() => setIsAddEventActive(false)}
+            className='bg-blue-500 hover:bg-blue-700 disabled:bg-gray-500 text-white font-bold py-2 px-5 rounded-xl ml-3 mb-3'
+          >
+            Close
+          </button>
+          {eventsNotInSeason.length !== 0 ? (
+            eventsNotInSeason.map((event) => {
```

Comment:
How are these events sorted? Ideally they should probably be sorted by `eventid` where highest eventid is at the top

Reply:
These are the events that are not part of a season and show up when you click on Add Events button. They're not sorted at all, they're not sorted by `eventid `but are sorted based on the order that events are added/removed from a season. However, if its your first time clicking the Add Event button when visiting a season page, they'll be order by how the API returns the data. Is there a reason this needs to be sorted by `eventid`?

Reply:
When adding events to a season, its usually best that most recently created events are at the top for ease of use -- rather than needing to scroll through possibly many events to find the event they're looking for. The higher the `eventid` the more recent that event was created

Reply:
Got it, will change that immediately!